src/prefect/core/flow.py

```diff
@@ -163,7 +166,7 @@ def __init__(
         self.result_handler = (
             result_handler or prefect.engine.get_default_result_handler_class()()
         )
-
+        self.has_result_handler = has_result_handler or bool(result_handler)
```

Comment:
So I totally see where you're going here; however, this does introduce a race condition wherein a user could claim their Flow has a result handler when in fact it does not.